sdk/kiosk/src/tx/kiosk.ts

```diff
@@ -335,35 +334,43 @@ export function returnValue(
  * Completes the full purchase flow that includes:
  * 1. Purchasing the item.
  * 2. Resolving all the transfer policies (if any).
- * 3. Returns the PurchasedItem OR places the item in the user's kiosk (if there's a kiosk lock policy).
+ * 3. Returns the item and whether the user can transfer it or not.
+ *
+ * If the item can be transferred, there's an extra transaction required by the user
+ * to transfer it to an address or place it in their kiosk.
  */
 export function purchaseAndResolvePolicies(
   tx: TransactionBlock,
   itemType: string,
   listing: KioskListing,
-  kioskId: string,
-  itemId: string,
+  kioskId: ObjectArgument,
+  item: ObjectArgument,
   policy: TransferPolicy,
-): TransactionArgument | null {
+  environment: RulesEnvironmentParam,
+  extraParams?: PurchaseOptionalParams,
+): PurchaseAndResolvePoliciesResponse {
   // if we don't pass the listing or the listing doens't have a price, return.
-  if (!listing || listing?.price === undefined) return null;
+  if (!listing || listing?.price === undefined)
```

Comment:
Let's just require `price`. Perhaps, instead we should require Coin... the reason for it is that developers might choose a custom coin and not use gas for it.

Reply:
Good point for the price! 
I like the `Coin` idea, I think though that this is a nice generic solution for 3-line purchase flows. 
Maybe we should support more `advanced use-case variations` for ideas like this, by possibly extending `extraParams` in the future to support `coin`, etc.